Adding nf-core sparsesignatures module

Katerina10-cloud opened the pull request:
## PR checklist

Closes #XXX <!-- If this PR fixes an issue, please link it here! -->

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

SPPearce commented:
> Due to the large container size and memory limitations in the CI environment, Docker and Singularity profile tests have been skipped. Only the Conda profile is tested in CI.

That's unusual, but reasonable. Can you confirm it does pass when using docker/singularity please

Katerina10-cloud commented:
In the latest commit, the Singularity profile test successfully passed in CI. You can check it on `Run nf-test/Test | singularity | 13 and 14`.

SPPearce commented on modules/nf-core/sparsesignatures/templates/main_script.R:
```diff
@@ -0,0 +1,223 @@
+# Load a reference SBS5 background signature from COSMIC
+data(background)
+
+seed_val <- if (!is.null(opt[["seed"]])) as.integer(opt[["seed"]]) else NULL
```
Is this setting a random seed?
If so, can we please set a default if not being set via args.

Katerina10-cloud replied:
I’ve updated the function to set a default seed `42` if none is provided via arguments. This ensures consistent behavior and avoids issues when seed is NULL.

Katerina10-cloud commented:
> I find it hard to understand what the base R `parse_args` code does, I'm sure that there is a better way to do that if you have tidyr/dplyr packages included into the container/conda environment. But presumably it works.

I’ve updated the `parse_args` function to use `dplyr/tidyr` functions, as suggested.

Katerina10-cloud commented on modules/nf-core/sparsesignatures/main.nf:
```diff
@@ -0,0 +1,46 @@
+    tuple val(meta), path("*_best_params_config.rds"), emit: signatures_bestConf_rds
+    tuple val(meta), path("*_nmf_Lasso_out.rds"), emit: signatures_nmfOut_rds
+    tuple val(meta), path("*_plot_all.rds"), emit: signatures_plot_rds
+    tuple val(meta), path("*_plot_all.pdf"), emit: signatures_plot_pdf
```
Fixed the alignment of the `emit:` statements.

Katerina10-cloud commented on modules/nf-core/sparsesignatures/meta.yml:
```diff
@@ -0,0 +1,118 @@
+  - "@elena-buscaroli"
+  - "@giorgiagandolfi"
+  - "@vvirgy"
+  - "@nicola-calonaci"
```
I’ve reduced the list of maintainers to only include the appropriate contributors for the nf-core module.

SPPearce commented on modules/nf-core/sparsesignatures/tests/nextflow.config:
Oh, could you please swap to using the newer `module_args` setup:
https://nf-co.re/docs/guidelines/components/modules#configuration-of-extargs-in-tests

SPPearce reviewed (commented):
I think this is virtually ready, apologies for not reviewing sooner.

Katerina10-cloud commented:
Hi @SPPearce,
I adressed your last comments and added GRCh37 genome test.

Katerina10-cloud commented:
> > > Can you add the v38 test back as well as the GRCh37 test please.
> > 
> > 
> > I’d like to clarify whether I can include both tests in the `main.nf.test` file?
> 
> Yes, please put both in the main.nf.test. Some modules have lots of tests (and this one has two, one with and without stub at the moment)

Now, I added both genome versions tests within `main.nf.test` and fixed labels in `main.nf`

SPPearce commented on modules/nf-core/sparsesignatures/main.nf:
```diff
@@ -0,0 +1,45 @@
+process SPARSE_SIGNATURES {
+    tag "$meta.id"
+    label "process_high"
```
Why did this go to high? That will default 12 cores and 72 GB RAM?
I don't think the module as written can use more than one core.

Katerina10-cloud replied:
Ok, since module includes cross validation step which is pretty long, I think "process_long" should be enough.
